packages/@aws-cdk/aws-ssm/lib/parameter.ts

```diff
@@ -0,0 +1,220 @@
+ */
+export abstract class Parameter extends cdk.Construct implements IParameter {
+  public abstract readonly parameterName: string;
+  public abstract readonly parameterType: string;
+
+  constructor(scope: cdk.Construct, id: string, _props: ParameterProps) {
+    super(scope, id);
+  }
+
+  public get parameterArn(): string {
+    return cdk.Stack.find(this).formatArn({
+      service: 'ssm',
+      resource: 'parameter',
+      resourceName: this.parameterName,
+    });
+  }
+
+  public grantRead(grantee: iam.IRole): IParameter {
+    grantee.addToPolicy(new iam.PolicyStatement()
+      .allow()
+      .addActions('ssm:DescribeParameters', 'ssm:GetParameter', 'ssm:GetParameterHistory')
+      .addResource(this.parameterArn));
+    return this;
+  }
+
+  public grantWrite(grantee: iam.IRole): IParameter {
+    grantee.addToPolicy(new iam.PolicyStatement()
+      .allow()
+      .addAction('ssm:PutParameter')
+      .addResource(this.parameterArn));
+    return this;
+  }
+}
+
+/**
+ * Creates a new String SSM Parameter.
+ */
+export class StringParameter extends Parameter implements IStringParameter {
+  public readonly parameterName: string;
+  public readonly parameterType: string;
+  public readonly parameterValue: string;
+
+  constructor(scope: cdk.Construct, id: string, props: StringParameterProps) {
+    super(scope, id, props);
+
+    if (props.allowedPattern) {
+      _assertValidValue(props.value, props.allowedPattern);
+    }
+
+    const resource = new ssm.CfnParameter(this, 'Resource', {
+      allowedPattern: props.allowedPattern,
+      description: props.description,
+      name: props.name,
+      type: 'String',
+      value: props.value,
+    });
+
+    this.parameterName = resource.parameterName;
+    this.parameterType = resource.parameterType;
+    this.parameterValue = resource.parameterValue;
+  }
+}
+
+/**
+ * Creates a new StringList SSM Parameter.
+ */
+export class StringListParameter extends Parameter implements IStringListParameter {
+  public readonly parameterName: string;
+  public readonly parameterType: string;
+  public readonly parameterValue: string[];
+
+  constructor(scope: cdk.Construct, id: string, props: StringListParameterProps) {
+    super(scope, id, props);
+
+    if (props.value.find(str => str.indexOf(',') !== -1)) {
+      throw new Error('Values of a StringList SSM Parameter cannot contain the \',\' character. Use a string parameter instead.');
+    }
+
+    if (props.allowedPattern) {
+      props.value.forEach(str => _assertValidValue(str, props.allowedPattern!));
+    }
+
+    const resource = new ssm.CfnParameter(this, 'Resource', {
+      allowedPattern: props.allowedPattern,
+      description: props.description,
+      name: props.name,
+      type: 'StringList',
+      value: props.value.join(','),
+    });
+
+    this.parameterName = resource.parameterName;
+    this.parameterType = resource.parameterType;
+    this.parameterValue = [resource.parameterValue];
```

Comment:
Waaaaitaminit. I don't think this is right.

Our L1 representation must be wrong; if the `parameterValue` is typed as a `"String"` this won't turn out as the correct value.

The L1 `parameterValue` must be untyped (or typed in a way that does not cause `cfn2ts` to already call `.toString()` or `.toList()`), so that it is represented as a `Token` on which you can call `.toList()` here.

Reply:
Wow, that's tricky. The L1 parameterValue is typed as `String` and for `StringList` the value is a comma separated list.

So this should be I believe:

```ts
this.parameterValue = Fn.split(',', resource.parameterValue); // this will return a string[]
```

Also make sure to verify this (manually) with an integration test...